internal/server/evaluation/evaluator.go

```diff
@@ -265,6 +191,137 @@ func (e *Evaluator) Evaluate(ctx context.Context, flag *flipt.Flag, r *flipt.Eva
 	return resp, nil
 }
 
+func (e *Evaluator) booleanMatch(r *rpcEvaluation.EvaluationRequest, flagValue bool, rollouts []*storage.EvaluationRollout) (*rpcEvaluation.BooleanEvaluationResponse, error) {
+	resp := &rpcEvaluation.BooleanEvaluationResponse{}
+
+	var lastRank int32
+
+	for _, rollout := range rollouts {
+		if rollout.Rank < lastRank {
+			return &rpcEvaluation.BooleanEvaluationResponse{}, errs.ErrInvalidf("rollout rank: %d detected out of order", rollout.Rank)
```

Comment:
💅 should be able to start returning `nil` again if we stop setting reason when error is non-nil.
Also, I think this should be and internal error as we control the rank field.
```suggestion
			return nil, fmt.Errorf("rollout rank: %d detected out of order", rollout.Rank)
```